mantidproject/mantid
# Create Hexahedral Detectors in memory for the StructuredDetector

The first iteration of the StructuredDetector implementation involved using the ShapeFactory to create Hexahedral shape objects using xml definitions. This proved to be quite an inefficient method for generating detector objects. This PR proposes a faster in-memory alternative to building the detector objects.

**To test:**

Code review and load LOKI definition (see attached, change extension to .xml)
[LOKI_Definition.txt](https://github.com/mantidproject/mantid/files/254671/LOKI_Definition.txt)

. This should now load even faster than before.

Fixes #15793.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
